src/extension.ts

```diff
@@ -205,17 +205,27 @@ async function autoUpdate() {
     }
 }
 
-function diagnosticCounter() {
-    let runningDiagnostics = 0;
+function progressCounter() {
+    const runningProgress: { [index: string]: boolean } = {};
+    const asPercent = (fraction: number): string => `${Math.round(fraction * 100)}%`;
     lc.onReady().then(() => {
-        lc.onNotification(new NotificationType('rustDocument/beginBuild'), function(_f: any) {
-            runningDiagnostics++;
-            startSpinner('RLS: working');
-        });
-        lc.onNotification(new NotificationType('rustDocument/diagnosticsEnd'), function(_f: any) {
-            runningDiagnostics--;
-            if (runningDiagnostics <= 0) {
-                stopSpinner('RLS: done');
+
+        stopSpinner('RLS');
+
+        lc.onNotification(new NotificationType('window/progress'), function (progress: any) {
+            if (progress.done) {
+                delete runningProgress[progress.id];
+            } else {
+                runningProgress[progress.id] = true;
+            }
+            if (Object.keys(runningProgress).length) {
+                const status =
+                    typeof progress.percentage === 'number' ? asPercent(progress.percentage) :
+                    progress.message ? progress.message :
```

Comment:
Title is ignored in favour of message, if it's specified - is that intended?
This leads to `RLS * [build]` followed by `RLS * crate_name` when building, where we lose information that we're building that crate.

Reply:
@Xanewok This is indeed intended. I wanted to keep the information flow at the bottom terse, and it's mostly obviously what it's doing. The `[<title>]` is to get something rather than `RLS *`, but could maybe dropped. 

It's kinda interesting to see how long it spends doing `[diagnostics]` though. Already now I can see we should look into performance here.